### ISSUE TITLE
Cannot use DiagnosticSuppressor when other analyzers are installed.

### ISSUE THREAD

**Issue by DasOhmoff** [dotnet/roslyn#44933]
Hello, in my code, my classes have a function called "Initialize". This function is called via reflection, and should only be allowed to be called via reflection, therefore I set it to private. When I do this I get the following Warning message: `IDE0051: Remove unused private members`. 

I want to tell through a Diagnostic Suppressor that this warning should be suppressed:
```
[assembly: SuppressMessage("CodeQuality", "IDE0051:Remove unused private members",
[DiagnosticAnalyzer(LanguageNames.CSharp)]
public class DiagnosticWarningSuppressor : DiagnosticSuppressor
{
    public override ImmutableArray<SuppressionDescriptor> SupportedSuppressions { get; } = ImmutableArray.Create(suppressionDescriptor);

    private const string Justification = "This method may be called via Reflection.";

    private static readonly SuppressionDescriptor suppressionDescriptor = new SuppressionDescriptor("CambicAnalyzer_InitializationCallViaReflection", "IDE0051", Justification);

    public override void ReportSuppressions(SuppressionAnalysisContext context)
    {
        foreach(Diagnostic diagnostic in context.ReportedDiagnostics)
        {
            SyntaxTree syntaxTree = diagnostic.Location.SourceTree;

            if(syntaxTree != null)
            {
                SyntaxNode rootNode   = syntaxTree.GetRoot(context.CancellationToken);
                SyntaxNode targetNode = rootNode.FindNode(diagnostic.Location.SourceSpan, false, true);

                SemanticModel semanticModel  = context.GetSemanticModel(syntaxTree);
                ISymbol       declaredSymbol = semanticModel.GetDeclaredSymbol(targetNode, context.CancellationToken);

                if(declaredSymbol is IMethodSymbol method && method.Name == "Initialize")
                    context.ReportSuppression(Suppression.Create(suppressionDescriptor, diagnostic));
            }
        }
    }
}
```

This works when no other analyzer is installed. But when another analyzer is installed, for example the "Microsoft.CodeAnalysis.FxCopAnalyzers". Then this does not work. How can I make the warning suppressor no matter what analyzers are installed?
